janitor/functions.py

```diff
@@ -1021,3 +1025,171 @@ def _round_to_fraction(number, denominator, digits=np.inf):
     df[colname] = df[colname].apply(_round_to_fraction_partial)
 
     return df
+
+
+@lru_cache(maxsize=32)
+def _convert_currency(
+    from_currency: str = None,
+    to_currency: str = None,
+    historical_date: date = None,
+):
+    """
+    Currency conversion for Pandas DataFrame column.
+    Helper function for `convert_currency` method.
+    The API used is: https://exchangeratesapi.io/
+    """
+
+    url = "https://api.exchangeratesapi.io"
+
+    if historical_date:
+        assert isinstance(historical_date, date) | isinstance(
+            historical_date, datetime
+        ), "historical_date must be a date or datetime object!"
+        if isinstance(historical_date, datetime):
+            assert historical_date >= datetime(
+                1999, 1, 4
+            ), "datetime must be later than 1999-01-04!"
+            string_date = str(historical_date)[:10]
+        else:
+            assert historical_date >= date(
+                1999, 1, 4
+            ), "date must be later than 1999-01-04!"
+            string_date = str(historical_date)
+        url = url + "/%s" % string_date
+    else:
+        url = url + "/latest"
+
+    currency_set = {
+        "AUD",
+        "BGN",
+        "BRL",
+        "CAD",
+        "CHF",
+        "CNY",
+        "CZK",
+        "DKK",
+        "EUR",
+        "GBP",
+        "HKD",
+        "HRK",
+        "HUF",
+        "IDR",
+        "ILS",
+        "INR",
+        "ISK",
+        "JPY",
+        "KRW",
+        "MXN",
+        "MYR",
+        "NOK",
+        "NZD",
+        "PHP",
+        "PLN",
+        "RON",
+        "RUB",
+        "SEK",
+        "SGD",
+        "THB",
+        "TRY",
+        "USD",
+        "ZAR",
+    }
+
+    assert (
```

Comment:
I'd like to suggest providing a `check_currency()` helper function.

```python
def check_currency(currency):
    if currency not in currency_set:
        raise ValueError(f"currency {currency} not in supported currency set, {currency_set}")
```

This would allow refactoring:

```suggestion
    check_currency(from_currency)
```

Reply:
I like it, but maybe call it `_ check_currency` to indicate that it's private?

Reply:
@szuckerman can do too! Though of course, there's no truly private functions in Python, but yes, the prefixed underscore is a good indicator.